red/api/editor/sshkeys.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+var express = require("express");
+var os = require("os");
+var runtime;
+var settings;
+var needsPermission = require("../auth").needsPermission;
+
+function getUsername(userObj) {
+    var username = os.hostname();
+    if ( userObj && userObj.name ) {
+        username = userObj.name;
+    }
+    return username;
+}
+
+module.exports = {
+    init: function(_runtime) {
+        runtime = _runtime;
+        settings = runtime.settings;
+    },
+    app: function() {
+        var app = express();
+
+        // SSH keys
+
+        // List all SSH keys
+        app.get("/", needsPermission("settings.read"), function(req,res) {
```

Comment:
I need to review the permissions for all of the project-level routes. I think there are cases where, if a user doesn't have write permission, they shouldn't be allowed to read it either. Given the sensitive nature of keys, I think this will be one of them.

No change needed here now... just a comment I wanted to share.